src/argilla/client/nightly.py

```diff
@@ -0,0 +1,367 @@
+        if isinstance(records, FeedbackRecord):
+            records = [records]
+
+        if self.schema is None:
+            warnings.warn("Since the `schema` hasn't been defined during the dataset creation, it will be inferred.")
+            self.schema = generate_pydantic_schema(records[0].fields)
+
+        for record in records:
+            record.fields = self.schema.parse_obj(record.fields)
+
+        # # If there are records already logged to Argilla, fetch one and get the schema
+        # self.schema = generate_pydantic_schema(self.fetch_one())
+        # # If there are no records logged to Argilla, check if `self.schema` has been set
+        # ...
+        # # If `self.schema` has not been set, just infer the schema based on the record
+        # ...
+        # record = record.dict() if isinstance(record, BaseModel) else record
+        self.client.add_records(
+            id=self.id,
+            records=[record.dict() for record in records],
+        )
+        # if self.__records is not None and isinstance(self.__records, list) and len(self.__records) > 0:
+        #     self.__records.append(record)
+
+    def fetch_one(self) -> "FeedbackItemModel":
+        if self.__records is None or len(self.__records) < 1:
+            return self.client.get_records(id=self.id, offset=0, limit=1).items[0]
+        return self.__records[0]
+
+    def iter(self, batch_size: int = 32) -> List["FeedbackItemModel"]:
+        if self.__records is None or len(self.__records) < 1:
+            first_batch = self.client.get_records(id=self.id, offset=0, limit=batch_size)
+            self.__records = first_batch.items
+            yield batch
+            total_batches = first_batch.total // batch_size
+            current_batch = 1
+            with tqdm(initial=current_batch, total=total_batches, desc="Fetching records from Argilla") as pbar:
+                while current_batch <= total_batches:
+                    batch = self.client.get_records(id=self.id, offset=batch_size, limit=batch_size)
+                    self.__records += batch.items
+                    yield batch
+                    current_batch += 1
+                    pbar.update(1)
+        else:
+            for batch in self.records[0::batch_size]:
+                yield batch
+
+
+def generate_pydantic_schema(
+    fields: Union[BaseModel, Dict[str, Any]], name: Optional[str] = "FieldsSchema"
+) -> BaseModel:
+    fields_schema = {key: (type(value), ...) for key, value in fields.items()}
+    return create_model(name, **fields_schema)
+
+
+def create_feedback_dataset(
+    name: str,
+    workspace: Optional[Union[str, rg.Workspace]] = None,
+    guidelines: Optional[str] = None,
+    fields: Optional[List[FieldSchema]] = None,
+    questions: Optional[List[QuestionSchema]] = None,
+) -> FeedbackDataset:
+    client = rg.active_client()
+
+    if workspace is None or isinstance(workspace, str):
+        try:
+            workspace = rg.Workspace.from_name(workspace)
```

Comment:
If the `workspace` is `None` this statement should fail. Could we use the `client.get_current_workspace()` to setup a default value?
```python
workspace = workspace or client.get_current_workspace()
```

Reply:
Indeed the `rg.Workspace.from_name` uses the default if the provided is None, so it won't fail 👍🏻

Reply:
Unless you prefer to set `name` in `rg.Workspace.from_name` as a mandatory value, and get the default in the line you mentioned instead.